packages/mds-geography-service/service/validators.ts

```diff
@@ -14,72 +14,82 @@
  * limitations under the License.
  */
 
-import { SchemaValidator, schemaValidator } from '@mds-core/mds-schema-validators'
+import { SchemaValidator } from '@mds-core/mds-schema-validators'
 import { UUID } from '@mds-core/mds-types'
+import Ajv from 'ajv'
 import gjv from 'geojson-validation'
-import Joi from 'joi'
 import {
   GeographyDomainCreateModel,
   GeographyMetadataDomainCreateModel,
   GetGeographiesOptions,
   GetPublishedGeographiesOptions
 } from '../@types'
 
+const ajvWithGeoJSON = new Ajv({ allErrors: true }).addKeyword({
+  keyword: 'valid-geojson',
+  schema: false,
+  validate: (geography_json: unknown) => {
+    try {
+      return gjv.valid(geography_json, false)
+    } catch (error) {
+      return false
+    }
+  }
+})
+
+const uuidSchema = { type: 'string', format: 'uuid' }
+
 export const { validate: validateGeographyDomainCreateModel, isValid: isValidGeographyDomainCreateModel } =
-  schemaValidator<GeographyDomainCreateModel>(
-    Joi.object<GeographyDomainCreateModel>()
-      .keys({
-        geography_id: Joi.string().uuid().required(),
-        name: Joi.string().max(255).allow(null),
-        description: Joi.string().max(255).allow(null),
-        effective_date: Joi.number().integer().allow(null),
-        publish_date: Joi.number().integer().allow(null),
-        prev_geographies: Joi.array().items(Joi.string().uuid()).allow(null),
-        geography_json: Joi.custom((geography_json, helpers) => {
-          try {
-            const [error] = gjv.valid(geography_json, true)
-            if (error !== undefined) {
-              return helpers.message({ custom: `GeoJSON is invalid ${error}` })
-            }
-          } catch (error) {
-            return helpers.message({ custom: `GeoJSON could not be validated` })
-          }
-          return geography_json
-        }).required()
-      })
-      .unknown(false)
+  SchemaValidator<GeographyDomainCreateModel>(
+    {
+      $id: 'Geography',
+      type: 'object',
+      properties: {
+        geography_id: uuidSchema,
+        name: { type: 'string', maxLength: 255 },
+        description: { type: 'string', maxLength: 255 },
+        effective_date: { type: 'integer' },
+        publish_date: { type: 'integer', nullable: true },
+        prev_geographies: { type: 'array', items: uuidSchema },
+        geography_json: { 'valid-geojson': true }
```

Comment:
Should default optional properties to `null`:
```suggestion
        geography_id: uuidSchema,
        name: { type: 'string', maxLength: 255, nullable: true, default: null },
        description: { type: 'string', maxLength: 255, nullable: true, default: null },
        effective_date: { type: 'integer', nullable: true, default: null },
        publish_date: { type: 'integer', nullable: true, nullable: true, default: null },
        prev_geographies: { type: 'array', items: uuidSchema, nullable: true, default: null },
        geography_json: { 'valid-geojson': true }
```